cuda/image_nonlinear_registration/nonlinear_registration.cu

```diff
@@ -108,12 +106,14 @@ int main(void)
 
     check_dif(dis,ddis);
 
-    std::cout << "\n==Emsumble TEST: nonlinear registration==" << std::endl;
+    std::cout << "\n==Ensemble TEST: nonlinear registration==" << std::endl;
 
     {
         tipl::time t("  nonlinear registration using cpu:");
         bool terminated = false;
-        dis.swap(decltype(dis)(dis.shape()));
+
```

Comment:
Here there was an issue that 'reference to non-const' must be an lvalue, whereas calling the constructor as previous makes it a temporary (not an lvalue). This caused a compile fail in GCC